File: src/main/java/com/server/computerscience/chatbot/dto/response/ChatGptResponseDto.java
package com.server.computerscience.chatbot.dto.response;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@ToString
public class ChatGptResponseDto {
	private String id;
	private String object;
	private long created;
	private String model;
	private Usage usage;
	private List<Choice> choices;

	@NoArgsConstructor
	@AllArgsConstructor
	@Getter
	@ToString
	public static class Choice {
		private Message message;
		private String finishReason;
		private int index;
	}

	@ToString
	@NoArgsConstructor
	@AllArgsConstructor
	@Getter
	public static class Message {
		private String role;
		private String content;
	}

	@ToString
	@NoArgsConstructor
	@AllArgsConstructor
	@Getter
	public static class Usage {
		private int promptTokens;
		private int completionTokens;
		private int totalTokens;
	}
}

File: core/src/main/kotlin/com/comssa/core/chatbot/dto/response/ChatGptResponseDto.kt
package com.comssa.core.chatbot.dto.response

data class ChatGptResponseDto(
	val id: String? = null,
	val `object`: String? = null,
	val created: Long = 0,
	val model: String? = null,
	val usage: Usage? = null,
	val choices: List<Choice>? = null,
) {
	val firstChoiceContent: String
		get() =
			this.choices
				?.get(0)
				?.message
				?.content ?: ""

	data class Choice(
		val message: Message? = null,
		val finishReason: String? = null,
		val index: Int = 0,
	)

	data class Message(
		val role: String? = null,
		val content: String? = null,
	)

	data class Usage(
		val promptTokens: Int = 0,
		val completionTokens: Int = 0,
		val totalTokens: Int = 0,
	)
}
